borg/helpers.py

```diff
@@ -332,11 +333,9 @@ def _match(self, path):
         raise NotImplementedError
 
 
-# For both PathPrefixPattern and FnmatchPattern, we require that
-# the pattern either match the whole path or an initial segment
-# of the path up to but not including a path separator.  To
-# unify the two cases, we add a path separator to the end of
-# the path before matching.
+# For both PathPrefixPattern, FnmatchPattern and ShellPattern, we require that the pattern either match the whole path
+# or an initial segment of the path up to but not including a path separator. To unify the two cases, we add a path
```

Comment:
> -both

Done.

> three cases?

Unless my English parser fails me the “two cases” refer to “either match the whole path or an initial segment […]”.